gensim/models/ldamodel.py

```diff
@@ -87,10 +87,11 @@ class LdaState(utils.SaveLoad):
 
     """
 
-    def __init__(self, eta, shape):
-        self.eta = eta
-        self.sstats = np.zeros(shape)
+    def __init__(self, eta, shape, dtype=np.float32):
+        self.eta = eta.astype(dtype, copy=False)
+        self.sstats = np.zeros(shape, dtype)
```

Comment:
Using positional arguments can lead to subtle bugs with numpy. Better use explicit names for keyword parameters: `dtype=dtype`.

Reply:
fixed